src/lib.rs

```diff
@@ -108,15 +108,16 @@ impl PublicKey {
     }
 
     /// Returns `true` if the signature matches the message.
-    pub fn verify<M: AsRef<[u8]>>(&self, sig: &Signature, msg: M) -> bool {
-        // TC
-        //self.verify_g2(sig, hash_g2(msg))
-        // BLST
-        let blst_sig = BlstSignature::from_bytes(&sig.to_bytes()).unwrap();
-        let blst_pk = BlstPublicKey::from_bytes(&self.to_bytes()).unwrap();
-        blst_sig.verify(false, msg.as_ref(), DST, &[], &blst_pk, false) == BLST_ERROR::BLST_SUCCESS
-        // DISABLED
-        //true
+    pub fn verify<M: AsRef<[u8]>>(&self, sig: &Signature, msg: M) -> Result<()> {
```

Comment:
This looks fine to me, but while it's open for change and consideration wanted to point out the ed25519-dalek `verify` function returns `-> Result<(), SignatureError>` (see [keypair.rs L255](https://github.com/dalek-cryptography/ed25519-dalek/blob/ad461f4f0f56507a2945510dedb914b3d3547c40/src/keypair.rs#L255)). Not saying this is necessarily better than `Result<()>` but wanted to make a note of it in case the difference has any significance or benefit for here.

Reply:
Right, I saw we have something similar [here](https://github.com/maidsafe/blsttc/blob/master/src/error.rs#L38) too, I mean, we have a single isolated error type instead of all errors being part of a single `Error` enum, which is what I'm doing with this PR, going to `Result<(), Error>` for all results instead of having some function returning a `Result<(), FromBytesError>`. I honestly don't know if it's worthwhile having separate error types, not sure either if there is some reason or explanation for benefits?